lib/configuration/variables/sources/json-parse.js

```diff
@@ -0,0 +1,30 @@
+'use strict';
+
+const ServerlessError = require('../../../serverless-error');
+const { get } = require('lodash');
+
+module.exports = {
+  resolve: ({ address, isSourceFulfilled, params }) => {
+    if (!params && params[0] == null) {
+      throw new ServerlessError(
+        'Missing JSON object or JSON string in variable "jsonParse" source',
```

Comment:
I did this to handle the return from the ssm package. 

Currently serverless when detecting a `{` in a securestring it will return the string already `JSON.parsed`. So the use case was tailored for our use of: abstracting a key inside the JSON object returned from SSM.

Thinking about it, it seems like this variable function is doing two things:
1. Parsing JSON (if needed)
2. Abstracting a specified key

#2 is what my use case needed b/c SSM already parses the JSON for me

Reply:
Do you mean, that when setting up a variable construct you're not sure whether SSM will return a JSON string or JSON object?

Reply:
Ok, so we can agree, that we will always know upfront wether we deal with JSON string or object, right?

It's not that SSM result is on SecureString on random basis

Reply:
Yes correct. When I call `jsonParse` I will know what data is going in and not expect something random

Reply:
Ok. For our use case I will still need a way to abstract the object key we need.

Do you suggest we add support to SSM to return the value of the object key we want? Or should we create a separate variable function to handle returning a single key?


I basically want to be able to do something like this:
```yaml
environment:
    MY_KEY: ${jsonParse(${ssm:path_to_secret}):my_key}
```

Where the value of `ssm:path_to_secret` will be a JSON object returned from ssm:
```javascript
{
  "my_key": "my_key_value"
}
```

Reply:
I guess alternatively we could have SSM NOT return a parsed JSON? But that's a breaking change that would require different handling

Reply:
That's a good point. In such case indeed it'll be useful to provide object pass through support to `jsonParse`.

At least I'm not sure if it's better to introduce another variable function, feels unnecessary complexity, and handling everything with `jsonParse` feels more elegant

Reply:
Just to confirm. We want to allow for both a JSON string and JSON object (for passthrough) in this function?

That would leave the biggest change to be getting rid of lodash usage?

Let me know!

Reply:
Yes, let's allow both object and string. Otherwise I believe my points remain valid. Thank you @celioreyes 👍